nilearn/regions/rena_clustering.py

```diff
@@ -153,6 +155,126 @@ def _make_edges_and_weights(X, mask_img):
     return edges, weights
 
 
+def _compute_weights_surface(X, mask, edges):
+    """Compute the weights for each edge using squared Euclidean distance.
+
+    Parameters
+    ----------
+    X : ndarray, shape = [n_samples, n_features]
+        Training data.
+
+    mask : boolean
+        Object used for masking the data.
+
```

Comment:
Samples are just the repetitions of observations or time points.

(Note that X here is a 2D array that we get after masking a SurfaceImage.)

Mask here is the initial boolean mask that was used to get X. So it would be False for vertices that should not be included and True for vertices that should be.

Meaning it would have the shape (1, n_vertices).